### PR TITLE
internal/ethapi: fix hex handling for eth_sign, personal_{sign,recover}

### PR DESCRIPTION
**Pull request by fjl**

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -289,14 +290,14 @@ func (s *PrivateAccountAPI) SendTransaction(ctx context.Context, args SendTxArgs
...
-func signHash(message string) []byte {
-	data := common.FromHex(message)
-	// Give context to the signed message. This prevents an adversery to sign a tx.
-	// It has no cryptographic purpose.
+// safely used to calculate a signature from.
+//
+// The hash is calulcated as
+//   keccak256("\x19Ethereum Signed Message:\n"${message length}${message}).
+//
+// This gives context to the signed message and an adversery to sign a tx.
```
Comment (holiman) [inline internal/ethapi/api.go]: "and an adversery to sign a tx." -> "and prevents an adversery from fooling a user to sign a tx"

Comment (fjl) [inline internal/ethapi/api.go]: It now says "prevents signing of transactions".

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -322,29 +318,20 @@ func (s *PrivateAccountAPI) Sign(ctx context.Context, message string, addr commo
...
 		return common.Address{}, fmt.Errorf("signature must be 65 bytes long")
 	}
-
 	// see crypto.Ecrecover description
 	if sig[64] == 27 || sig[64] == 28 {
 		sig[64] -= 27
 	}
-
-	rpk, err := crypto.Ecrecover(hash, sig)
+	rpk, err := crypto.Ecrecover(signHash(data), sig)
```
Comment (fjl) [inline internal/ethapi/api.go]: The documentation for ecrecover says

```text
// Note, this function is compatible with eth_sign and personal_sign. As such it recovers
// the address of:
// hash = keccak256("\x19Ethereum Signed Message:\n"${message length}${message})
// addr = ecrecover(hash, signature)
```

Comment (fjl) [inline internal/ethapi/api.go]: Not sure what else needs to be added.

Comment (bas-vk) [inline internal/ethapi/api.go]: Sry, your are right. Didn't expand the documentation and missed exactly that part.

LGTM

**Pull request merged**